src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableActorContext.cs

```diff
@@ -0,0 +1,112 @@
+        /// A logger for logging information about this actor.
+        /// </summary>
+        ILogger Logger { get; }
+
+        /// <summary>
+        /// Whether this actor is freshly constructed, i.e. did not exist prior to this operation being called.
+        /// </summary>
+        bool NewlyConstructed { get; }
+
+        /// <summary>
+        /// Gets a typed view of the state, by deserializing the JSON.
+        /// </summary>
+        /// <typeparam name="TState">the JSON-serializable type of the actor state.</typeparam>
+        /// <returns>a typed view that allows reading and updating.</returns>
+        IStateView<TState> GetStateAs<TState>(Formatting formatting = Formatting.Indented, JsonSerializerSettings settings = null);
+
+        /// <summary>
+        /// Gets the content that was passed to this operation as a deserialized value.
+        /// </summary>
+        /// <typeparam name="T">The JSON-serializable type used for the operation content.</typeparam>
+        /// <returns>the operation content, or default(<typeparamref name="T"/>) if none.</returns>
+        T GetOperationContent<T>();
+
+        /// <summary>
+        /// Gets the content that was passed to this operation as a deserialized value.
+        /// </summary>
+        /// <param name="contentType">The JSON-serializable type used for the operation content.</param>
+        /// <returns>the operation content, or default(<paramref name="contentType"/>) if none.</returns>
+        object GetOperationContent(Type contentType);
+
+        /// <summary>
+        /// Returns the given result to the caller of this operation. 
+        /// </summary>
+        /// <typeparam name="TResult">the type of the result.</typeparam>
+        /// <param name="result">the result to return.</param>
+        void Return<TResult>(TResult result);
+
+        /// <summary>
+        /// Returns the given result to the caller of this operation. 
+        /// </summary>
+        /// <param name="result">the result to return.</param>
+        /// <param name="resultType">optionally, the type of the result.</param>
+        void Return(object result, Type resultType = null);
+
+        /// <summary>
+        /// Delete this actor after this operation completes.
+        /// </summary>
+        void DestructOnExit();
+    }
+
+    /// <summary>
+    /// A typed view of the current state of the actor.
+    /// </summary>
+    /// <typeparam name="TState">The JSON-serializable type used for this actor.</typeparam>
+    public interface IStateView<TState> : IStateView
+    {
+        /// <summary>
+        /// The current state of the actor.
+        /// </summary>
+        TState Value { get; set; }
+    }
+
+    /// <summary>
+    /// A view of the current state of the actor.
+    /// </summary>
+    public interface IStateView : IDisposable
+    {
+        /// <summary>
+        /// Serializes the current state to JSON.
+        /// </summary>
+        void WriteBack();
+    }
```

Comment:
I suggest putting both of these interfaces into their own file, `IStateView.cs`. That's the recommendation of the .NET Framework design guidelines.